{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Network Assembly Notebook"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "from copy import copy\n",
    "from tools import networkHelpers as nh\n",
    "from importlib import reload\n",
    "%matplotlib inline"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Get HGNC Mapper"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "hgnc_mapper = nh.fetch_hgnc_mapper()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Get Uniprot mapper"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "('P31946', 'YWHAB')"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "uni_mapper = pd.read_csv('../data/networkInputs/HUMAN_9606_idmapping.dat', sep='\\t', header=None)\n",
    "uni_mapper = uni_mapper[uni_mapper[1] == 'Gene_Name'].set_index(0)[2].to_dict()\n",
    "\n",
    "next(iter(uni_mapper.items()))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "146363\n",
      "26468\n"
     ]
    }
   ],
   "source": [
    "# size info\n",
    "uni_rev = {y:x for x,y in uni_mapper.items()}\n",
    "print(len(uni_mapper))\n",
    "print(len(uni_rev))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Extra mappers from UniProt, e.g. ENSG"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "#!wget ftp://ftp.uniprot.org/pub/databases/uniprot/current_release/knowledgebase/idmapping/by_organism/HUMAN_9606_idmapping_selected.tab.gz\n",
    "#!mv HUMAN_9606_idmapping_selected.tab.gz ../data/networkInputs/\n",
    "#!gunzip ../data/networkInputs/HUMAN_9606_idmapping_selected.tab.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(169389, 22)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "      <th>9</th>\n",
       "      <th>...</th>\n",
       "      <th>12</th>\n",
       "      <th>13</th>\n",
       "      <th>14</th>\n",
       "      <th>15</th>\n",
       "      <th>16</th>\n",
       "      <th>17</th>\n",
       "      <th>18</th>\n",
       "      <th>19</th>\n",
       "      <th>20</th>\n",
       "      <th>21</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>P31946</td>\n",
       "      <td>1433B_HUMAN</td>\n",
       "      <td>7529</td>\n",
       "      <td>NP_003395.1; NP_647539.1; XP_016883528.1</td>\n",
       "      <td>4507949; 377656702; 67464628; 1345590; 1034625...</td>\n",
       "      <td>2BQ0:A; 2BQ0:B; 2C23:A; 4DNK:A; 4DNK:B; 5N10:A...</td>\n",
       "      <td>GO:0005737; GO:0005829; GO:0070062; GO:0005925...</td>\n",
       "      <td>UniRef100_P31946</td>\n",
       "      <td>UniRef90_P31946</td>\n",
       "      <td>UniRef50_P31946</td>\n",
       "      <td>...</td>\n",
       "      <td>9606</td>\n",
       "      <td>601289</td>\n",
       "      <td>Hs.643544</td>\n",
       "      <td>8515476; 14702039; 11780052; 15489334; 2357255...</td>\n",
       "      <td>X57346; AK292717; AL008725; CH471077; CH471077...</td>\n",
       "      <td>CAA40621.1; BAF85406.1; -; EAW75893.1; EAW7589...</td>\n",
       "      <td>ENSG00000166913</td>\n",
       "      <td>ENST00000353703; ENST00000372839</td>\n",
       "      <td>ENSP00000300161; ENSP00000361930</td>\n",
       "      <td>11996670; 12364343; 12437930; 12468542; 124825...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>P62258</td>\n",
       "      <td>1433E_HUMAN</td>\n",
       "      <td>7531</td>\n",
       "      <td>NP_006752.1</td>\n",
       "      <td>374074368; 62131678; 67464424; 194379794; 2210...</td>\n",
       "      <td>2BR9:A; 3UAL:A; 3UBW:A; 6EIH:A</td>\n",
       "      <td>GO:0090724; GO:0005737; GO:0005829; GO:0070062...</td>\n",
       "      <td>UniRef100_P62258</td>\n",
       "      <td>UniRef90_P62258</td>\n",
       "      <td>UniRef50_P62258</td>\n",
       "      <td>...</td>\n",
       "      <td>9606</td>\n",
       "      <td>605066</td>\n",
       "      <td>Hs.513851</td>\n",
       "      <td>7644510; 8858348; 8684458; 20417184; 14702039;...</td>\n",
       "      <td>U20972; U54778; U43399; U43430; U28936; AB0171...</td>\n",
       "      <td>AAC50175.1; AAC50710.1; AAC50625.1; AAD00026.1...</td>\n",
       "      <td>ENSG00000108953; ENSG00000274474</td>\n",
       "      <td>ENST00000264335; ENST00000571732; ENST00000616...</td>\n",
       "      <td>ENSP00000264335; ENSP00000461762; ENSP00000481...</td>\n",
       "      <td>15838597; 11782387; 12657644; 14966136; 153649...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Q04917</td>\n",
       "      <td>1433F_HUMAN</td>\n",
       "      <td>7533</td>\n",
       "      <td>NP_003396.1</td>\n",
       "      <td>4507951; 437363; 83754686; 83754699; 83754700;...</td>\n",
       "      <td>2C63:A; 2C63:B; 2C63:C; 2C63:D; 2C74:A; 2C74:B</td>\n",
       "      <td>GO:0005737; GO:0005829; GO:0070062; GO:0098978...</td>\n",
       "      <td>UniRef100_Q04917</td>\n",
       "      <td>UniRef90_Q04917</td>\n",
       "      <td>UniRef50_P62258</td>\n",
       "      <td>...</td>\n",
       "      <td>9606</td>\n",
       "      <td>113508</td>\n",
       "      <td>Hs.226755</td>\n",
       "      <td>8218406; 1578511; 8561965; 8812417; 15461802; ...</td>\n",
       "      <td>L20422; X80536; X78138; X57345; D78577; S80794...</td>\n",
       "      <td>AAA35483.1; CAA56676.1; CAA55017.1; CAA40620.1...</td>\n",
       "      <td>ENSG00000128245</td>\n",
       "      <td>ENST00000248975</td>\n",
       "      <td>ENSP00000248975</td>\n",
       "      <td>10206237; 11121172; 11996670; 12176995; 124801...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>P61981</td>\n",
       "      <td>1433G_HUMAN</td>\n",
       "      <td>7532</td>\n",
       "      <td>NP_036611.2</td>\n",
       "      <td>6016838; 380764684; 635576381; 82407956; 82407...</td>\n",
       "      <td>2B05:A; 2B05:B; 2B05:C; 2B05:D; 2B05:E; 2B05:F...</td>\n",
       "      <td>GO:0005829; GO:0070062; GO:0005925; GO:0016020...</td>\n",
       "      <td>UniRef100_P61981</td>\n",
       "      <td>UniRef90_P61981</td>\n",
       "      <td>UniRef50_P61981</td>\n",
       "      <td>...</td>\n",
       "      <td>9606</td>\n",
       "      <td>605356; 617665</td>\n",
       "      <td>Hs.744840</td>\n",
       "      <td>10433554; 10486217; 12853948; 15489334; 235725...</td>\n",
       "      <td>AF142498; AB024334; CR541904; CR541925; AC0063...</td>\n",
       "      <td>AAD48408.1; BAA85184.1; CAG46702.1; CAG46723.1...</td>\n",
       "      <td>ENSG00000170027</td>\n",
       "      <td>ENST00000307630</td>\n",
       "      <td>ENSP00000306330</td>\n",
       "      <td>11824616; 11996670; 12364343; 12482592; 150572...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>P31947</td>\n",
       "      <td>1433S_HUMAN</td>\n",
       "      <td>2810</td>\n",
       "      <td>NP_006133.1</td>\n",
       "      <td>436408756; 306991738; 969812714; 350610438; 35...</td>\n",
       "      <td>1YWT:A; 1YWT:B; 1YZ5:A; 1YZ5:B; 3IQJ:A; 3IQU:A...</td>\n",
       "      <td>GO:0005829; GO:0070062; GO:0005615; GO:0005739...</td>\n",
       "      <td>UniRef100_P31947</td>\n",
       "      <td>UniRef90_P31947</td>\n",
       "      <td>UniRef50_P61981</td>\n",
       "      <td>...</td>\n",
       "      <td>9606</td>\n",
       "      <td>601290</td>\n",
       "      <td>Hs.523718</td>\n",
       "      <td>1390337; 8515476; 9659898; 16710414; 15489334;...</td>\n",
       "      <td>M93010; X57348; AF029081; AF029082; CR541905; ...</td>\n",
       "      <td>AAA59546.1; CAA40623.1; AAC52029.1; AAC52030.1...</td>\n",
       "      <td>ENSG00000175793</td>\n",
       "      <td>ENST00000339276</td>\n",
       "      <td>ENSP00000340989</td>\n",
       "      <td>10969776; 12582028; 12730237; 12787309; 145172...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 22 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       0            1     2                                         3   \\\n",
       "0  P31946  1433B_HUMAN  7529  NP_003395.1; NP_647539.1; XP_016883528.1   \n",
       "1  P62258  1433E_HUMAN  7531                               NP_006752.1   \n",
       "2  Q04917  1433F_HUMAN  7533                               NP_003396.1   \n",
       "3  P61981  1433G_HUMAN  7532                               NP_036611.2   \n",
       "4  P31947  1433S_HUMAN  2810                               NP_006133.1   \n",
       "\n",
       "                                                  4   \\\n",
       "0  4507949; 377656702; 67464628; 1345590; 1034625...   \n",
       "1  374074368; 62131678; 67464424; 194379794; 2210...   \n",
       "2  4507951; 437363; 83754686; 83754699; 83754700;...   \n",
       "3  6016838; 380764684; 635576381; 82407956; 82407...   \n",
       "4  436408756; 306991738; 969812714; 350610438; 35...   \n",
       "\n",
       "                                                  5   \\\n",
       "0  2BQ0:A; 2BQ0:B; 2C23:A; 4DNK:A; 4DNK:B; 5N10:A...   \n",
       "1                     2BR9:A; 3UAL:A; 3UBW:A; 6EIH:A   \n",
       "2     2C63:A; 2C63:B; 2C63:C; 2C63:D; 2C74:A; 2C74:B   \n",
       "3  2B05:A; 2B05:B; 2B05:C; 2B05:D; 2B05:E; 2B05:F...   \n",
       "4  1YWT:A; 1YWT:B; 1YZ5:A; 1YZ5:B; 3IQJ:A; 3IQU:A...   \n",
       "\n",
       "                                                  6                 7   \\\n",
       "0  GO:0005737; GO:0005829; GO:0070062; GO:0005925...  UniRef100_P31946   \n",
       "1  GO:0090724; GO:0005737; GO:0005829; GO:0070062...  UniRef100_P62258   \n",
       "2  GO:0005737; GO:0005829; GO:0070062; GO:0098978...  UniRef100_Q04917   \n",
       "3  GO:0005829; GO:0070062; GO:0005925; GO:0016020...  UniRef100_P61981   \n",
       "4  GO:0005829; GO:0070062; GO:0005615; GO:0005739...  UniRef100_P31947   \n",
       "\n",
       "                8                9   ...    12              13         14  \\\n",
       "0  UniRef90_P31946  UniRef50_P31946  ...  9606          601289  Hs.643544   \n",
       "1  UniRef90_P62258  UniRef50_P62258  ...  9606          605066  Hs.513851   \n",
       "2  UniRef90_Q04917  UniRef50_P62258  ...  9606          113508  Hs.226755   \n",
       "3  UniRef90_P61981  UniRef50_P61981  ...  9606  605356; 617665  Hs.744840   \n",
       "4  UniRef90_P31947  UniRef50_P61981  ...  9606          601290  Hs.523718   \n",
       "\n",
       "                                                  15  \\\n",
       "0  8515476; 14702039; 11780052; 15489334; 2357255...   \n",
       "1  7644510; 8858348; 8684458; 20417184; 14702039;...   \n",
       "2  8218406; 1578511; 8561965; 8812417; 15461802; ...   \n",
       "3  10433554; 10486217; 12853948; 15489334; 235725...   \n",
       "4  1390337; 8515476; 9659898; 16710414; 15489334;...   \n",
       "\n",
       "                                                  16  \\\n",
       "0  X57346; AK292717; AL008725; CH471077; CH471077...   \n",
       "1  U20972; U54778; U43399; U43430; U28936; AB0171...   \n",
       "2  L20422; X80536; X78138; X57345; D78577; S80794...   \n",
       "3  AF142498; AB024334; CR541904; CR541925; AC0063...   \n",
       "4  M93010; X57348; AF029081; AF029082; CR541905; ...   \n",
       "\n",
       "                                                  17  \\\n",
       "0  CAA40621.1; BAF85406.1; -; EAW75893.1; EAW7589...   \n",
       "1  AAC50175.1; AAC50710.1; AAC50625.1; AAD00026.1...   \n",
       "2  AAA35483.1; CAA56676.1; CAA55017.1; CAA40620.1...   \n",
       "3  AAD48408.1; BAA85184.1; CAG46702.1; CAG46723.1...   \n",
       "4  AAA59546.1; CAA40623.1; AAC52029.1; AAC52030.1...   \n",
       "\n",
       "                                 18  \\\n",
       "0                   ENSG00000166913   \n",
       "1  ENSG00000108953; ENSG00000274474   \n",
       "2                   ENSG00000128245   \n",
       "3                   ENSG00000170027   \n",
       "4                   ENSG00000175793   \n",
       "\n",
       "                                                  19  \\\n",
       "0                   ENST00000353703; ENST00000372839   \n",
       "1  ENST00000264335; ENST00000571732; ENST00000616...   \n",
       "2                                    ENST00000248975   \n",
       "3                                    ENST00000307630   \n",
       "4                                    ENST00000339276   \n",
       "\n",
       "                                                  20  \\\n",
       "0                   ENSP00000300161; ENSP00000361930   \n",
       "1  ENSP00000264335; ENSP00000461762; ENSP00000481...   \n",
       "2                                    ENSP00000248975   \n",
       "3                                    ENSP00000306330   \n",
       "4                                    ENSP00000340989   \n",
       "\n",
       "                                                  21  \n",
       "0  11996670; 12364343; 12437930; 12468542; 124825...  \n",
       "1  15838597; 11782387; 12657644; 14966136; 153649...  \n",
       "2  10206237; 11121172; 11996670; 12176995; 124801...  \n",
       "3  11824616; 11996670; 12364343; 12482592; 150572...  \n",
       "4  10969776; 12582028; 12730237; 12787309; 145172...  \n",
       "\n",
       "[5 rows x 22 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "t = pd.read_csv('../data/networkInputs/HUMAN_9606_idmapping_selected.tab', sep='\\t', header=None, low_memory=False)\n",
    "display(t.shape)\n",
    "display(t.head())\n",
    "\n",
    "entrez_to_uniprot_number = t.set_index(2)[0].to_dict()\n",
    "ensg_to_uniprot_number = t.set_index(18)[0].to_dict()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Kinase Master List (to filter)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Uniprot Protein</th>\n",
       "      <th>MS Gene</th>\n",
       "      <th>RNAseq Gene</th>\n",
       "      <th>RNAseq Accession</th>\n",
       "      <th>Family</th>\n",
       "      <th>Mouse Uniprot Protein</th>\n",
       "      <th>Mouse RNAseq gene</th>\n",
       "      <th>Mouse RNAseq Accession</th>\n",
       "      <th>Kinome Render Tree Name</th>\n",
       "      <th>Aliases (Conservative)</th>\n",
       "      <th>...</th>\n",
       "      <th>Aliases</th>\n",
       "      <th>description</th>\n",
       "      <th>other_designations</th>\n",
       "      <th>Entrez_Symbol</th>\n",
       "      <th>Old_Name</th>\n",
       "      <th>Entrez_Synonyms</th>\n",
       "      <th>Entrez_other_designations</th>\n",
       "      <th>Unnamed: 18</th>\n",
       "      <th>Gene Symbol</th>\n",
       "      <th>Gene Symbol and Synonyms</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AAK1</td>\n",
       "      <td>AAK1</td>\n",
       "      <td>AAK1</td>\n",
       "      <td>22848.0</td>\n",
       "      <td>Other</td>\n",
       "      <td>Aak1</td>\n",
       "      <td>Aak1</td>\n",
       "      <td>269774</td>\n",
       "      <td>AAK1</td>\n",
       "      <td>AAK1,</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AP2 associated kinase 1</td>\n",
       "      <td>adaptor-associated kinase 1</td>\n",
       "      <td>AAK1</td>\n",
       "      <td>AAK1</td>\n",
       "      <td>KIAA1048|MGC138170</td>\n",
       "      <td>adaptor-associated kinase 1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AAK1</td>\n",
       "      <td>AAK1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AAPK1</td>\n",
       "      <td>PRKAA1</td>\n",
       "      <td>PRKAA1</td>\n",
       "      <td>5562.0</td>\n",
       "      <td>CAMK</td>\n",
       "      <td>Aapk1</td>\n",
       "      <td>Prkaa1</td>\n",
       "      <td>105787</td>\n",
       "      <td>AMPK[alpha]1</td>\n",
       "      <td>PRKAA1, AMPKa1</td>\n",
       "      <td>...</td>\n",
       "      <td>AMPK, AMPKa1</td>\n",
       "      <td>protein kinase, AMP-activated, alpha 1 catalyt...</td>\n",
       "      <td>5'-AMP-activated protein kinase, catalytic alp...</td>\n",
       "      <td>PRKAA1</td>\n",
       "      <td>AMPKa1</td>\n",
       "      <td>AMPK|AMPKa1|MGC33776|MGC57364</td>\n",
       "      <td>5'-AMP-activated protein kinase, catalytic alp...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PRKAA1</td>\n",
       "      <td>AMPK, PRKAA1, AMPKa1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AAPK2</td>\n",
       "      <td>PRKAA2</td>\n",
       "      <td>PRKAA2</td>\n",
       "      <td>5563.0</td>\n",
       "      <td>CAMK</td>\n",
       "      <td>Aapk2</td>\n",
       "      <td>Prkaa2</td>\n",
       "      <td>108079</td>\n",
       "      <td>AMPK[alpha]2</td>\n",
       "      <td>PRKAA2, AMPK2, AMPKa2, PRKAA</td>\n",
       "      <td>...</td>\n",
       "      <td>AMPK, AMPK2, AMPKa2, PRKAA</td>\n",
       "      <td>protein kinase, AMP-activated, alpha 2 catalyt...</td>\n",
       "      <td>5'-AMP-activated protein kinase, catalytic alp...</td>\n",
       "      <td>PRKAA2</td>\n",
       "      <td>AMPKa2</td>\n",
       "      <td>AMPK|AMPK2|PRKAA</td>\n",
       "      <td>5'-AMP-activated protein kinase, catalytic alp...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PRKAA2</td>\n",
       "      <td>AMPK2, AMPK, PRKAA, AMPKa2, PRKAA2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ABL1</td>\n",
       "      <td>ABL1</td>\n",
       "      <td>ABL1</td>\n",
       "      <td>25.0</td>\n",
       "      <td>TK</td>\n",
       "      <td>Abl1</td>\n",
       "      <td>Abl1</td>\n",
       "      <td>11350</td>\n",
       "      <td>Abl</td>\n",
       "      <td>ABL1,ABL, JTK7, bcr/abl, c-ABL, c-ABL1, v-abl</td>\n",
       "      <td>...</td>\n",
       "      <td>ABL, JTK7, bcr/abl, c-ABL, c-ABL1, p150, v-abl</td>\n",
       "      <td>ABL proto-oncogene 1, non-receptor tyrosine ki...</td>\n",
       "      <td>Abelson tyrosine-protein kinase 1|bcr/c-abl on...</td>\n",
       "      <td>ABL1</td>\n",
       "      <td>ABL</td>\n",
       "      <td>ABL|JTK7|c-ABL|p150|v-abl</td>\n",
       "      <td>Abelson murine leukemia viral (v-abl) oncogene...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ABL1</td>\n",
       "      <td>c-ABL1, p150, ABL, JTK7, v-abl, bcr/abl, c-ABL...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ABL2</td>\n",
       "      <td>ABL2</td>\n",
       "      <td>ABL2</td>\n",
       "      <td>27.0</td>\n",
       "      <td>TK</td>\n",
       "      <td>Abl2</td>\n",
       "      <td>Abl2</td>\n",
       "      <td>11352</td>\n",
       "      <td>Arg</td>\n",
       "      <td>ABL2,ABLL, ARG</td>\n",
       "      <td>...</td>\n",
       "      <td>ABLL, ARG</td>\n",
       "      <td>ABL proto-oncogene 2, non-receptor tyrosine ki...</td>\n",
       "      <td>abelson-related gene protein|c-abl oncogene 2,...</td>\n",
       "      <td>ABL2</td>\n",
       "      <td>ARG</td>\n",
       "      <td>ABLL|ARG</td>\n",
       "      <td>Abelson murine leukemia viral (v-abl) oncogene...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ABL2</td>\n",
       "      <td>ABL2, ABLL, ARG</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  Uniprot Protein MS Gene RNAseq Gene  RNAseq Accession Family  \\\n",
       "0            AAK1    AAK1        AAK1           22848.0  Other   \n",
       "1           AAPK1  PRKAA1      PRKAA1            5562.0   CAMK   \n",
       "2           AAPK2  PRKAA2      PRKAA2            5563.0   CAMK   \n",
       "3            ABL1    ABL1        ABL1              25.0     TK   \n",
       "4            ABL2    ABL2        ABL2              27.0     TK   \n",
       "\n",
       "  Mouse Uniprot Protein Mouse RNAseq gene Mouse RNAseq Accession  \\\n",
       "0                  Aak1              Aak1                 269774   \n",
       "1                 Aapk1            Prkaa1                 105787   \n",
       "2                 Aapk2            Prkaa2                 108079   \n",
       "3                  Abl1              Abl1                  11350   \n",
       "4                  Abl2              Abl2                  11352   \n",
       "\n",
       "  Kinome Render Tree Name                         Aliases (Conservative)  ...  \\\n",
       "0                    AAK1                                          AAK1,  ...   \n",
       "1            AMPK[alpha]1                                 PRKAA1, AMPKa1  ...   \n",
       "2            AMPK[alpha]2                   PRKAA2, AMPK2, AMPKa2, PRKAA  ...   \n",
       "3                     Abl  ABL1,ABL, JTK7, bcr/abl, c-ABL, c-ABL1, v-abl  ...   \n",
       "4                     Arg                                 ABL2,ABLL, ARG  ...   \n",
       "\n",
       "                                          Aliases  \\\n",
       "0                                             NaN   \n",
       "1                                    AMPK, AMPKa1   \n",
       "2                      AMPK, AMPK2, AMPKa2, PRKAA   \n",
       "3  ABL, JTK7, bcr/abl, c-ABL, c-ABL1, p150, v-abl   \n",
       "4                                       ABLL, ARG   \n",
       "\n",
       "                                         description  \\\n",
       "0                            AP2 associated kinase 1   \n",
       "1  protein kinase, AMP-activated, alpha 1 catalyt...   \n",
       "2  protein kinase, AMP-activated, alpha 2 catalyt...   \n",
       "3  ABL proto-oncogene 1, non-receptor tyrosine ki...   \n",
       "4  ABL proto-oncogene 2, non-receptor tyrosine ki...   \n",
       "\n",
       "                                  other_designations Entrez_Symbol Old_Name  \\\n",
       "0                        adaptor-associated kinase 1          AAK1     AAK1   \n",
       "1  5'-AMP-activated protein kinase, catalytic alp...        PRKAA1   AMPKa1   \n",
       "2  5'-AMP-activated protein kinase, catalytic alp...        PRKAA2   AMPKa2   \n",
       "3  Abelson tyrosine-protein kinase 1|bcr/c-abl on...          ABL1      ABL   \n",
       "4  abelson-related gene protein|c-abl oncogene 2,...          ABL2      ARG   \n",
       "\n",
       "                 Entrez_Synonyms  \\\n",
       "0             KIAA1048|MGC138170   \n",
       "1  AMPK|AMPKa1|MGC33776|MGC57364   \n",
       "2               AMPK|AMPK2|PRKAA   \n",
       "3      ABL|JTK7|c-ABL|p150|v-abl   \n",
       "4                       ABLL|ARG   \n",
       "\n",
       "                           Entrez_other_designations Unnamed: 18  Gene Symbol  \\\n",
       "0                        adaptor-associated kinase 1         NaN         AAK1   \n",
       "1  5'-AMP-activated protein kinase, catalytic alp...         NaN       PRKAA1   \n",
       "2  5'-AMP-activated protein kinase, catalytic alp...         NaN       PRKAA2   \n",
       "3  Abelson murine leukemia viral (v-abl) oncogene...         NaN         ABL1   \n",
       "4  Abelson murine leukemia viral (v-abl) oncogene...         NaN         ABL2   \n",
       "\n",
       "                            Gene Symbol and Synonyms  \n",
       "0                                               AAK1  \n",
       "1                               AMPK, PRKAA1, AMPKa1  \n",
       "2                 AMPK2, AMPK, PRKAA, AMPKa2, PRKAA2  \n",
       "3  c-ABL1, p150, ABL, JTK7, v-abl, bcr/abl, c-ABL...  \n",
       "4                                    ABL2, ABLL, ARG  \n",
       "\n",
       "[5 rows x 21 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "kmast = pd.read_excel('../data/ref/KINASESmasterlist_w_Aliases.xlsx')\n",
    "kmast.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "kinome_set = set(hgnc_mapper[x] for x in kmast['MS Gene'] if 'GYK' != x and 'SGK494' != x)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "570\n",
      "499\n",
      "744\n"
     ]
    }
   ],
   "source": [
    "print(len(set(kmast['Uniprot Protein'])))\n",
    "print(len(set(kmast['Symbol'])))\n",
    "print(len(set(kmast['Symbol'])|set(kmast['Uniprot Protein'])))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Start importing sources"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "data_sources = {}\n",
    "data_preproc = {}\n",
    "\n",
    "# hardcode MAPK3 due to MAPKAPK3 issues\n",
    "id_dict = lambda x: hgnc_mapper[x]  if x in hgnc_mapper.keys()\\\n",
    "                                        else 'MAPK3' if x == 'P27361'\\\n",
    "                                        else hgnc_mapper[uni_mapper[x]] if x in uni_mapper.keys() and uni_mapper[x] in hgnc_mapper.keys()\\\n",
    "                                        else None\n",
    "                                        "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Int64Index([0, 1, 2, 3, 4, 5], dtype='int64')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AL1A1_HUMAN</td>\n",
       "      <td>216</td>\n",
       "      <td>AL1A1_HUMAN</td>\n",
       "      <td>216</td>\n",
       "      <td>0.76</td>\n",
       "      <td>experiments:in vivo,Two-hybrid;pmids:12081471,...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ITA7_HUMAN</td>\n",
       "      <td>3679</td>\n",
       "      <td>ACHA_HUMAN</td>\n",
       "      <td>1134</td>\n",
       "      <td>0.73</td>\n",
       "      <td>experiments:in vivo,Affinity Capture-Western,a...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NEB1_HUMAN</td>\n",
       "      <td>55607</td>\n",
       "      <td>ACTG_HUMAN</td>\n",
       "      <td>71</td>\n",
       "      <td>0.65</td>\n",
       "      <td>experiments:in vitro,in vivo;pmids:9362513,120...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SRGN_HUMAN</td>\n",
       "      <td>5552</td>\n",
       "      <td>CD44_HUMAN</td>\n",
       "      <td>960</td>\n",
       "      <td>0.63</td>\n",
       "      <td>experiments:in vivo;pmids:9334256,16189514,167...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GRB7_HUMAN</td>\n",
       "      <td>2886</td>\n",
       "      <td>ERBB2_HUMAN</td>\n",
       "      <td>2064</td>\n",
       "      <td>0.90</td>\n",
       "      <td>experiments:in vitro,in vivo,Reconstituted Com...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             0      1            2     3     4  \\\n",
       "0  AL1A1_HUMAN    216  AL1A1_HUMAN   216  0.76   \n",
       "1   ITA7_HUMAN   3679   ACHA_HUMAN  1134  0.73   \n",
       "2   NEB1_HUMAN  55607   ACTG_HUMAN    71  0.65   \n",
       "3   SRGN_HUMAN   5552   CD44_HUMAN   960  0.63   \n",
       "4   GRB7_HUMAN   2886  ERBB2_HUMAN  2064  0.90   \n",
       "\n",
       "                                                   5  \n",
       "0  experiments:in vivo,Two-hybrid;pmids:12081471,...  \n",
       "1  experiments:in vivo,Affinity Capture-Western,a...  \n",
       "2  experiments:in vitro,in vivo;pmids:9362513,120...  \n",
       "3  experiments:in vivo;pmids:9334256,16189514,167...  \n",
       "4  experiments:in vitro,in vivo,Reconstituted Com...  "
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hippie = pd.read_csv('../data/networkInputs/hippie_current.txt', sep='\\t', header=None)\n",
    "\n",
    "data_sources['hippie'] = hippie\n",
    "data_preproc['hippie'] = {\n",
    "    'columns':[0,2],\n",
    "    'maps':[\n",
    "        lambda x: id_dict(x.split('_')[0]),\n",
    "        lambda x: id_dict(x.split('_')[0])\n",
    "    ]\n",
    "}\n",
    "\n",
    "print(hippie.columns)\n",
    "hippie.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Index(['GENE', 'KINASE', 'KIN_ACC_ID', 'KIN_ORGANISM', 'SUBSTRATE',\n",
      "       'SUB_GENE_ID', 'SUB_ACC_ID', 'SUB_GENE', 'SUB_ORGANISM', 'SUB_MOD_RSD',\n",
      "       'SITE_GRP_ID', 'SITE_+/-7_AA', 'DOMAIN', 'IN_VIVO_RXN', 'IN_VITRO_RXN',\n",
      "       'CST_CAT#'],\n",
      "      dtype='object')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GENE</th>\n",
       "      <th>KINASE</th>\n",
       "      <th>KIN_ACC_ID</th>\n",
       "      <th>KIN_ORGANISM</th>\n",
       "      <th>SUBSTRATE</th>\n",
       "      <th>SUB_GENE_ID</th>\n",
       "      <th>SUB_ACC_ID</th>\n",
       "      <th>SUB_GENE</th>\n",
       "      <th>SUB_ORGANISM</th>\n",
       "      <th>SUB_MOD_RSD</th>\n",
       "      <th>SITE_GRP_ID</th>\n",
       "      <th>SITE_+/-7_AA</th>\n",
       "      <th>DOMAIN</th>\n",
       "      <th>IN_VIVO_RXN</th>\n",
       "      <th>IN_VITRO_RXN</th>\n",
       "      <th>CST_CAT#</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Pak2</td>\n",
       "      <td>PAK2</td>\n",
       "      <td>Q64303</td>\n",
       "      <td>rat</td>\n",
       "      <td>MEK1</td>\n",
       "      <td>170851.0</td>\n",
       "      <td>Q01986</td>\n",
       "      <td>Map2k1</td>\n",
       "      <td>rat</td>\n",
       "      <td>S298</td>\n",
       "      <td>448284</td>\n",
       "      <td>RtPGRPLsSYGMDSR</td>\n",
       "      <td>Pkinase</td>\n",
       "      <td></td>\n",
       "      <td>X</td>\n",
       "      <td>9128; 98195</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Pak2</td>\n",
       "      <td>PAK2</td>\n",
       "      <td>Q64303</td>\n",
       "      <td>rat</td>\n",
       "      <td>PRKD1</td>\n",
       "      <td>85421.0</td>\n",
       "      <td>Q9WTQ1</td>\n",
       "      <td>Prkd1</td>\n",
       "      <td>rat</td>\n",
       "      <td>S203</td>\n",
       "      <td>449896</td>\n",
       "      <td>GVRRRRLsNVsLTGL</td>\n",
       "      <td>NaN</td>\n",
       "      <td>X</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Pak2</td>\n",
       "      <td>PAK2</td>\n",
       "      <td>Q64303</td>\n",
       "      <td>rat</td>\n",
       "      <td>prolactin</td>\n",
       "      <td>5617.0</td>\n",
       "      <td>P01236</td>\n",
       "      <td>PRL</td>\n",
       "      <td>human</td>\n",
       "      <td>S207</td>\n",
       "      <td>451732</td>\n",
       "      <td>LHCLRRDsHKIDNYL</td>\n",
       "      <td>Hormone_1</td>\n",
       "      <td></td>\n",
       "      <td>X</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Pak2</td>\n",
       "      <td>PAK2</td>\n",
       "      <td>Q64303</td>\n",
       "      <td>rat</td>\n",
       "      <td>prolactin</td>\n",
       "      <td>24683.0</td>\n",
       "      <td>P01237</td>\n",
       "      <td>Prl</td>\n",
       "      <td>rat</td>\n",
       "      <td>S206</td>\n",
       "      <td>451732</td>\n",
       "      <td>IRCLRRDsHKVDNYL</td>\n",
       "      <td>Hormone_1</td>\n",
       "      <td></td>\n",
       "      <td>X</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>EIF2AK1</td>\n",
       "      <td>HRI</td>\n",
       "      <td>Q9BQI3</td>\n",
       "      <td>human</td>\n",
       "      <td>eIF2-alpha</td>\n",
       "      <td>54318.0</td>\n",
       "      <td>P68101</td>\n",
       "      <td>Eif2s1</td>\n",
       "      <td>rat</td>\n",
       "      <td>S52</td>\n",
       "      <td>447635</td>\n",
       "      <td>MILLSELsRRRIRSI</td>\n",
       "      <td>S1</td>\n",
       "      <td></td>\n",
       "      <td>X</td>\n",
       "      <td>3597; 9721; 3398; 5199</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      GENE KINASE KIN_ACC_ID KIN_ORGANISM   SUBSTRATE  SUB_GENE_ID SUB_ACC_ID  \\\n",
       "0     Pak2   PAK2     Q64303          rat        MEK1     170851.0     Q01986   \n",
       "1     Pak2   PAK2     Q64303          rat       PRKD1      85421.0     Q9WTQ1   \n",
       "2     Pak2   PAK2     Q64303          rat   prolactin       5617.0     P01236   \n",
       "3     Pak2   PAK2     Q64303          rat   prolactin      24683.0     P01237   \n",
       "4  EIF2AK1    HRI     Q9BQI3        human  eIF2-alpha      54318.0     P68101   \n",
       "\n",
       "  SUB_GENE SUB_ORGANISM SUB_MOD_RSD  SITE_GRP_ID     SITE_+/-7_AA     DOMAIN  \\\n",
       "0   Map2k1          rat        S298       448284  RtPGRPLsSYGMDSR    Pkinase   \n",
       "1    Prkd1          rat        S203       449896  GVRRRRLsNVsLTGL        NaN   \n",
       "2      PRL        human        S207       451732  LHCLRRDsHKIDNYL  Hormone_1   \n",
       "3      Prl          rat        S206       451732  IRCLRRDsHKVDNYL  Hormone_1   \n",
       "4   Eif2s1          rat         S52       447635  MILLSELsRRRIRSI         S1   \n",
       "\n",
       "  IN_VIVO_RXN IN_VITRO_RXN                CST_CAT#  \n",
       "0                        X             9128; 98195  \n",
       "1           X                                  NaN  \n",
       "2                        X                     NaN  \n",
       "3                        X                     NaN  \n",
       "4                        X  3597; 9721; 3398; 5199  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "phosphosite_substrate = pd.read_csv('../data/networkInputs/Kinase_Substrate_Dataset.txt', sep='\\t', skiprows=2)\n",
    "\n",
    "data_sources['phosphosite_substrate'] = phosphosite_substrate\n",
    "data_preproc['phosphosite_substrate'] = {\n",
    "    'columns':['GENE','SUB_GENE'],\n",
    "    'maps':[\n",
    "        lambda x: id_dict(x.upper()),\n",
    "        lambda x: id_dict(x.upper())\n",
    "    ]\n",
    "}\n",
    "\n",
    "print(phosphosite_substrate.columns)\n",
    "display(phosphosite_substrate.head())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Int64Index([0, 1, 2, 3, 4, 5, 6, 7], dtype='int64')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ALDH1A1</td>\n",
       "      <td>1</td>\n",
       "      <td>NP_000680.2</td>\n",
       "      <td>ALDH1A1</td>\n",
       "      <td>1</td>\n",
       "      <td>NP_000680.2</td>\n",
       "      <td>in vivo;yeast 2-hybrid</td>\n",
       "      <td>1,208,147,116,189,510</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ITGA7</td>\n",
       "      <td>2761</td>\n",
       "      <td>NP_001138468.1</td>\n",
       "      <td>CHRNA1</td>\n",
       "      <td>7</td>\n",
       "      <td>NP_001034612.1</td>\n",
       "      <td>in vivo</td>\n",
       "      <td>10910772</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PPP1R9A</td>\n",
       "      <td>16000</td>\n",
       "      <td>NP_060120.2</td>\n",
       "      <td>ACTG1</td>\n",
       "      <td>17</td>\n",
       "      <td>NP_001605.1</td>\n",
       "      <td>in vitro;in vivo</td>\n",
       "      <td>936,251,312,052,877</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SRGN</td>\n",
       "      <td>1513</td>\n",
       "      <td>NP_002718.2</td>\n",
       "      <td>CD44</td>\n",
       "      <td>115</td>\n",
       "      <td>NP_000601.3</td>\n",
       "      <td>in vivo</td>\n",
       "      <td>9334256</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GRB7</td>\n",
       "      <td>3311</td>\n",
       "      <td>NP_005301.2</td>\n",
       "      <td>ERBB2</td>\n",
       "      <td>1281</td>\n",
       "      <td>NP_004439.2</td>\n",
       "      <td>in vitro;in vivo</td>\n",
       "      <td>9079677</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         0      1               2        3     4               5  \\\n",
       "0  ALDH1A1      1     NP_000680.2  ALDH1A1     1     NP_000680.2   \n",
       "1    ITGA7   2761  NP_001138468.1   CHRNA1     7  NP_001034612.1   \n",
       "2  PPP1R9A  16000     NP_060120.2    ACTG1    17     NP_001605.1   \n",
       "3     SRGN   1513     NP_002718.2     CD44   115     NP_000601.3   \n",
       "4     GRB7   3311     NP_005301.2    ERBB2  1281     NP_004439.2   \n",
       "\n",
       "                        6                      7  \n",
       "0  in vivo;yeast 2-hybrid  1,208,147,116,189,510  \n",
       "1                 in vivo               10910772  \n",
       "2        in vitro;in vivo    936,251,312,052,877  \n",
       "3                 in vivo                9334256  \n",
       "4        in vitro;in vivo                9079677  "
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hprd = pd.read_csv('../data/networkInputs/HPRD_ALL_BINARY_PROTEIN_PROTEIN_INTERACTIONS.txt', sep='\\t', header=None)\n",
    "\n",
    "data_sources['hprd'] = hprd\n",
    "data_preproc['hprd'] = {\n",
    "    'columns':[0,3],\n",
    "    'maps':[\n",
    "        lambda x: id_dict(x.upper()),\n",
    "        lambda x: id_dict(x.upper())\n",
    "    ]\n",
    "}\n",
    "\n",
    "print(hprd.columns)\n",
    "hprd.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Index(['# Interactor 1 uniprot id', 'Interactor 1 Ensembl gene id',\n",
      "       'Interactor 1 Entrez Gene id', 'Interactor 2 uniprot id',\n",
      "       'Interactor 2 Ensembl gene id', 'Interactor 2 Entrez Gene id',\n",
      "       'Interaction type', 'Interaction context', 'Pubmed references'],\n",
      "      dtype='object')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th># Interactor 1 uniprot id</th>\n",
       "      <th>Interactor 1 Ensembl gene id</th>\n",
       "      <th>Interactor 1 Entrez Gene id</th>\n",
       "      <th>Interactor 2 uniprot id</th>\n",
       "      <th>Interactor 2 Ensembl gene id</th>\n",
       "      <th>Interactor 2 Entrez Gene id</th>\n",
       "      <th>Interaction type</th>\n",
       "      <th>Interaction context</th>\n",
       "      <th>Pubmed references</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>uniprotkb:Q9Y287</td>\n",
       "      <td>ENSEMBL:ENSG00000136156</td>\n",
       "      <td>entrezgene/locuslink:9445</td>\n",
       "      <td>uniprotkb:Q9Y287</td>\n",
       "      <td>ENSEMBL:ENSG00000136156</td>\n",
       "      <td>entrezgene/locuslink:9445</td>\n",
       "      <td>physical association</td>\n",
       "      <td>reactome:R-HSA-976871</td>\n",
       "      <td>14690516|10391242</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>uniprotkb:P37840</td>\n",
       "      <td>ENSEMBL:ENSG00000145335</td>\n",
       "      <td>entrezgene/locuslink:6622</td>\n",
       "      <td>uniprotkb:P37840</td>\n",
       "      <td>ENSEMBL:ENSG00000145335</td>\n",
       "      <td>entrezgene/locuslink:6622</td>\n",
       "      <td>physical association</td>\n",
       "      <td>reactome:R-HSA-1247852</td>\n",
       "      <td>24243840</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>uniprotkb:P0DJI8</td>\n",
       "      <td>ENSEMBL:ENSG00000173432</td>\n",
       "      <td>entrezgene/locuslink:6288</td>\n",
       "      <td>uniprotkb:P0DJI8</td>\n",
       "      <td>ENSEMBL:ENSG00000173432</td>\n",
       "      <td>entrezgene/locuslink:6288</td>\n",
       "      <td>physical association</td>\n",
       "      <td>reactome:R-HSA-976898</td>\n",
       "      <td>19393650|103558</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>uniprotkb:P06727</td>\n",
       "      <td>ENSEMBL:ENSG00000110244</td>\n",
       "      <td>entrezgene/locuslink:337</td>\n",
       "      <td>uniprotkb:P06727</td>\n",
       "      <td>ENSEMBL:ENSG00000110244</td>\n",
       "      <td>entrezgene/locuslink:337</td>\n",
       "      <td>physical association</td>\n",
       "      <td>reactome:R-HSA-976889</td>\n",
       "      <td>15146166</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>uniprotkb:P01160</td>\n",
       "      <td>ENSEMBL:ENSG00000175206</td>\n",
       "      <td>entrezgene/locuslink:4878</td>\n",
       "      <td>uniprotkb:P01160</td>\n",
       "      <td>ENSEMBL:ENSG00000175206</td>\n",
       "      <td>entrezgene/locuslink:4878</td>\n",
       "      <td>physical association</td>\n",
       "      <td>reactome:R-HSA-976987</td>\n",
       "      <td>2142465|2945573</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  # Interactor 1 uniprot id Interactor 1 Ensembl gene id  \\\n",
       "0          uniprotkb:Q9Y287      ENSEMBL:ENSG00000136156   \n",
       "1          uniprotkb:P37840      ENSEMBL:ENSG00000145335   \n",
       "2          uniprotkb:P0DJI8      ENSEMBL:ENSG00000173432   \n",
       "3          uniprotkb:P06727      ENSEMBL:ENSG00000110244   \n",
       "4          uniprotkb:P01160      ENSEMBL:ENSG00000175206   \n",
       "\n",
       "  Interactor 1 Entrez Gene id Interactor 2 uniprot id  \\\n",
       "0   entrezgene/locuslink:9445        uniprotkb:Q9Y287   \n",
       "1   entrezgene/locuslink:6622        uniprotkb:P37840   \n",
       "2   entrezgene/locuslink:6288        uniprotkb:P0DJI8   \n",
       "3    entrezgene/locuslink:337        uniprotkb:P06727   \n",
       "4   entrezgene/locuslink:4878        uniprotkb:P01160   \n",
       "\n",
       "  Interactor 2 Ensembl gene id Interactor 2 Entrez Gene id  \\\n",
       "0      ENSEMBL:ENSG00000136156   entrezgene/locuslink:9445   \n",
       "1      ENSEMBL:ENSG00000145335   entrezgene/locuslink:6622   \n",
       "2      ENSEMBL:ENSG00000173432   entrezgene/locuslink:6288   \n",
       "3      ENSEMBL:ENSG00000110244    entrezgene/locuslink:337   \n",
       "4      ENSEMBL:ENSG00000175206   entrezgene/locuslink:4878   \n",
       "\n",
       "       Interaction type     Interaction context  Pubmed references  \n",
       "0  physical association   reactome:R-HSA-976871  14690516|10391242  \n",
       "1  physical association  reactome:R-HSA-1247852           24243840  \n",
       "2  physical association   reactome:R-HSA-976898    19393650|103558  \n",
       "3  physical association   reactome:R-HSA-976889           15146166  \n",
       "4  physical association   reactome:R-HSA-976987    2142465|2945573  "
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "reactome = pd.read_csv('../data/networkInputs/reactome.homo_sapiens.interactions.tab-delimited.txt', sep='\\t')\n",
    "\n",
    "data_sources['reactome'] = reactome\n",
    "data_preproc['reactome'] = {\n",
    "    'columns': ['# Interactor 1 uniprot id','Interactor 2 uniprot id'],\n",
    "    'maps': [\n",
    "        lambda x: id_dict(x.split(':')[1]),\n",
    "        lambda x: id_dict(x.split(':')[1])\n",
    "    ]\n",
    "}\n",
    "\n",
    "\n",
    "print(reactome.columns)\n",
    "reactome.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "#!wget https://downloads.thebiogrid.org/Download/BioGRID/Release-Archive/BIOGRID-3.5.169/BIOGRID-ORGANISM-3.5.169.tab2.zip\n",
    "#!unzip -q BIOGRID-ORGANISM-3.5.169.tab2.zip\n",
    "#!mv BIOGRID-ORGANISM-Homo_sapiens-3.5.169.tab2.txt ../data/networkInputs/\n",
    "#!rm `ls | grep 'BIOGRID-ORGANISM'`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Index(['#BioGRID Interaction ID', 'Entrez Gene Interactor A',\n",
      "       'Entrez Gene Interactor B', 'BioGRID ID Interactor A',\n",
      "       'BioGRID ID Interactor B', 'Systematic Name Interactor A',\n",
      "       'Systematic Name Interactor B', 'Official Symbol Interactor A',\n",
      "       'Official Symbol Interactor B', 'Synonyms Interactor A',\n",
      "       'Synonyms Interactor B', 'Experimental System',\n",
      "       'Experimental System Type', 'Author', 'Pubmed ID',\n",
      "       'Organism Interactor A', 'Organism Interactor B', 'Throughput', 'Score',\n",
      "       'Modification', 'Phenotypes', 'Qualifications', 'Tags',\n",
      "       'Source Database'],\n",
      "      dtype='object')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#BioGRID Interaction ID</th>\n",
       "      <th>Entrez Gene Interactor A</th>\n",
       "      <th>Entrez Gene Interactor B</th>\n",
       "      <th>BioGRID ID Interactor A</th>\n",
       "      <th>BioGRID ID Interactor B</th>\n",
       "      <th>Systematic Name Interactor A</th>\n",
       "      <th>Systematic Name Interactor B</th>\n",
       "      <th>Official Symbol Interactor A</th>\n",
       "      <th>Official Symbol Interactor B</th>\n",
       "      <th>Synonyms Interactor A</th>\n",
       "      <th>...</th>\n",
       "      <th>Pubmed ID</th>\n",
       "      <th>Organism Interactor A</th>\n",
       "      <th>Organism Interactor B</th>\n",
       "      <th>Throughput</th>\n",
       "      <th>Score</th>\n",
       "      <th>Modification</th>\n",
       "      <th>Phenotypes</th>\n",
       "      <th>Qualifications</th>\n",
       "      <th>Tags</th>\n",
       "      <th>Source Database</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>103</td>\n",
       "      <td>6416</td>\n",
       "      <td>2318</td>\n",
       "      <td>112315</td>\n",
       "      <td>108607</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>MAP2K4</td>\n",
       "      <td>FLNC</td>\n",
       "      <td>JNKK|JNKK1|MAPKK4|MEK4|MKK4|PRKMK4|SAPKK-1|SAP...</td>\n",
       "      <td>...</td>\n",
       "      <td>9006895</td>\n",
       "      <td>9606</td>\n",
       "      <td>9606</td>\n",
       "      <td>Low Throughput</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>BIOGRID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>117</td>\n",
       "      <td>84665</td>\n",
       "      <td>88</td>\n",
       "      <td>124185</td>\n",
       "      <td>106603</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>MYPN</td>\n",
       "      <td>ACTN2</td>\n",
       "      <td>CMD1DD|CMH22|MYOP|RCM4</td>\n",
       "      <td>...</td>\n",
       "      <td>11309420</td>\n",
       "      <td>9606</td>\n",
       "      <td>9606</td>\n",
       "      <td>Low Throughput</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>BIOGRID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>183</td>\n",
       "      <td>90</td>\n",
       "      <td>2339</td>\n",
       "      <td>106605</td>\n",
       "      <td>108625</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>ACVR1</td>\n",
       "      <td>FNTA</td>\n",
       "      <td>ACTRI|ACVR1A|ACVRLK2|ALK2|FOP|SKR1|TSRI</td>\n",
       "      <td>...</td>\n",
       "      <td>8599089</td>\n",
       "      <td>9606</td>\n",
       "      <td>9606</td>\n",
       "      <td>Low Throughput</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>BIOGRID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>278</td>\n",
       "      <td>2624</td>\n",
       "      <td>5371</td>\n",
       "      <td>108894</td>\n",
       "      <td>111384</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>GATA2</td>\n",
       "      <td>PML</td>\n",
       "      <td>DCML|IMD21|MONOMAC|NFE1B</td>\n",
       "      <td>...</td>\n",
       "      <td>10938104</td>\n",
       "      <td>9606</td>\n",
       "      <td>9606</td>\n",
       "      <td>Low Throughput</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>BIOGRID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>418</td>\n",
       "      <td>6118</td>\n",
       "      <td>6774</td>\n",
       "      <td>112038</td>\n",
       "      <td>112651</td>\n",
       "      <td>RP4-547C9.3</td>\n",
       "      <td>-</td>\n",
       "      <td>RPA2</td>\n",
       "      <td>STAT3</td>\n",
       "      <td>REPA2|RP-A p32|RP-A p34|RPA32</td>\n",
       "      <td>...</td>\n",
       "      <td>10875894</td>\n",
       "      <td>9606</td>\n",
       "      <td>9606</td>\n",
       "      <td>Low Throughput</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>BIOGRID</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 24 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   #BioGRID Interaction ID  Entrez Gene Interactor A  \\\n",
       "0                      103                      6416   \n",
       "1                      117                     84665   \n",
       "2                      183                        90   \n",
       "3                      278                      2624   \n",
       "4                      418                      6118   \n",
       "\n",
       "   Entrez Gene Interactor B  BioGRID ID Interactor A  BioGRID ID Interactor B  \\\n",
       "0                      2318                   112315                   108607   \n",
       "1                        88                   124185                   106603   \n",
       "2                      2339                   106605                   108625   \n",
       "3                      5371                   108894                   111384   \n",
       "4                      6774                   112038                   112651   \n",
       "\n",
       "  Systematic Name Interactor A Systematic Name Interactor B  \\\n",
       "0                            -                            -   \n",
       "1                            -                            -   \n",
       "2                            -                            -   \n",
       "3                            -                            -   \n",
       "4                  RP4-547C9.3                            -   \n",
       "\n",
       "  Official Symbol Interactor A Official Symbol Interactor B  \\\n",
       "0                       MAP2K4                         FLNC   \n",
       "1                         MYPN                        ACTN2   \n",
       "2                        ACVR1                         FNTA   \n",
       "3                        GATA2                          PML   \n",
       "4                         RPA2                        STAT3   \n",
       "\n",
       "                               Synonyms Interactor A  ... Pubmed ID  \\\n",
       "0  JNKK|JNKK1|MAPKK4|MEK4|MKK4|PRKMK4|SAPKK-1|SAP...  ...   9006895   \n",
       "1                             CMD1DD|CMH22|MYOP|RCM4  ...  11309420   \n",
       "2            ACTRI|ACVR1A|ACVRLK2|ALK2|FOP|SKR1|TSRI  ...   8599089   \n",
       "3                           DCML|IMD21|MONOMAC|NFE1B  ...  10938104   \n",
       "4                      REPA2|RP-A p32|RP-A p34|RPA32  ...  10875894   \n",
       "\n",
       "  Organism Interactor A Organism Interactor B      Throughput  Score  \\\n",
       "0                  9606                  9606  Low Throughput      -   \n",
       "1                  9606                  9606  Low Throughput      -   \n",
       "2                  9606                  9606  Low Throughput      -   \n",
       "3                  9606                  9606  Low Throughput      -   \n",
       "4                  9606                  9606  Low Throughput      -   \n",
       "\n",
       "   Modification  Phenotypes Qualifications Tags Source Database  \n",
       "0             -           -              -    -         BIOGRID  \n",
       "1             -           -              -    -         BIOGRID  \n",
       "2             -           -              -    -         BIOGRID  \n",
       "3             -           -              -    -         BIOGRID  \n",
       "4             -           -              -    -         BIOGRID  \n",
       "\n",
       "[5 rows x 24 columns]"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "biogrid = pd.read_csv('../data/networkInputs/BIOGRID-ORGANISM-Homo_sapiens-3.5.169.tab2.txt', low_memory=False, sep='\\t')\n",
    "\n",
    "data_sources['biogrid'] = biogrid\n",
    "data_preproc['biogrid'] = {\n",
    "    'columns': ['Official Symbol Interactor A','Official Symbol Interactor B'],\n",
    "    'maps': [\n",
    "        lambda x: id_dict(x),\n",
    "        lambda x: id_dict(x)\n",
    "    ]\n",
    "}\n",
    "\n",
    "print(biogrid.columns)\n",
    "biogrid.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Index(['Dataset', 'SwissProt1', 'SwissProt2'], dtype='object')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Dataset</th>\n",
       "      <th>SwissProt1</th>\n",
       "      <th>SwissProt2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>SOURAV_MAPK_LOW</td>\n",
       "      <td>P63000</td>\n",
       "      <td>A0AUZ9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>IntAct</td>\n",
       "      <td>Q96CV9</td>\n",
       "      <td>A0AUZ9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BioGrid</td>\n",
       "      <td>P0CG48</td>\n",
       "      <td>A0AV96</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>IntAct_Mouse</td>\n",
       "      <td>P62258</td>\n",
       "      <td>A0AV96</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>IntAct_Mouse</td>\n",
       "      <td>P63104</td>\n",
       "      <td>A0AV96</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           Dataset SwissProt1 SwissProt2\n",
       "0  SOURAV_MAPK_LOW     P63000     A0AUZ9\n",
       "1           IntAct     Q96CV9     A0AUZ9\n",
       "2          BioGrid     P0CG48     A0AV96\n",
       "3     IntAct_Mouse     P62258     A0AV96\n",
       "4     IntAct_Mouse     P63104     A0AV96"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "i2d = pd.read_csv('../data/networkInputs/i2d.2_9.txt', sep='\\t')\n",
    "\n",
    "data_sources['i2d'] = i2d\n",
    "data_preproc['i2d'] = {\n",
    "    'columns': ['SwissProt1','SwissProt2'],\n",
    "    'maps': [\n",
    "        lambda x: id_dict(x),\n",
    "        lambda x: id_dict(x)\n",
    "    ]\n",
    "}\n",
    "\n",
    "print(i2d.columns)\n",
    "i2d.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Int64Index([0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14], dtype='int64')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "      <th>9</th>\n",
       "      <th>10</th>\n",
       "      <th>11</th>\n",
       "      <th>12</th>\n",
       "      <th>13</th>\n",
       "      <th>14</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>uniprotkb:Q72547</td>\n",
       "      <td>uniprotkb:Q72547</td>\n",
       "      <td>intact:EBI-7484755|intact:MINT-8208544</td>\n",
       "      <td>intact:EBI-7484755|intact:MINT-8208544</td>\n",
       "      <td>psi-mi:q72547_9hiv1(display_long)|uniprotkb:po...</td>\n",
       "      <td>psi-mi:q72547_9hiv1(display_long)|uniprotkb:po...</td>\n",
       "      <td>psi-mi:\"MI:0030\"(cross-linking study)</td>\n",
       "      <td>Nishitsuji et al. (2011)</td>\n",
       "      <td>pubmed:22004763|imex:IM-16791|mint:MINT-820698...</td>\n",
       "      <td>taxid:11676(9hiv1)|taxid:11676(Human immunodef...</td>\n",
       "      <td>taxid:11676(9hiv1)|taxid:11676(Human immunodef...</td>\n",
       "      <td>psi-mi:\"MI:0915\"(physical association)</td>\n",
       "      <td>psi-mi:\"MI:0471\"(MINT)</td>\n",
       "      <td>intact:EBI-7484777|mint:MINT-8206988|imex:IM-1...</td>\n",
       "      <td>intact-miscore:0.40</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>uniprotkb:Q72547</td>\n",
       "      <td>uniprotkb:Q72547</td>\n",
       "      <td>intact:EBI-7484755|intact:MINT-8208544</td>\n",
       "      <td>intact:EBI-7484755|intact:MINT-8208544</td>\n",
       "      <td>psi-mi:q72547_9hiv1(display_long)|uniprotkb:po...</td>\n",
       "      <td>psi-mi:q72547_9hiv1(display_long)|uniprotkb:po...</td>\n",
       "      <td>psi-mi:\"MI:0030\"(cross-linking study)</td>\n",
       "      <td>Nishitsuji et al. (2011)</td>\n",
       "      <td>pubmed:22004763|imex:IM-16791|mint:MINT-820698...</td>\n",
       "      <td>taxid:11676(9hiv1)|taxid:11676(Human immunodef...</td>\n",
       "      <td>taxid:11676(9hiv1)|taxid:11676(Human immunodef...</td>\n",
       "      <td>psi-mi:\"MI:0915\"(physical association)</td>\n",
       "      <td>psi-mi:\"MI:0471\"(MINT)</td>\n",
       "      <td>intact:EBI-7484916|mint:MINT-8207003|imex:IM-1...</td>\n",
       "      <td>intact-miscore:0.40</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>uniprotkb:Q72547</td>\n",
       "      <td>uniprotkb:Q72547</td>\n",
       "      <td>intact:EBI-7484755|intact:MINT-8208544</td>\n",
       "      <td>intact:EBI-7484755|intact:MINT-8208544</td>\n",
       "      <td>psi-mi:q72547_9hiv1(display_long)|uniprotkb:po...</td>\n",
       "      <td>psi-mi:q72547_9hiv1(display_long)|uniprotkb:po...</td>\n",
       "      <td>psi-mi:\"MI:0030\"(cross-linking study)</td>\n",
       "      <td>Nishitsuji et al. (2011)</td>\n",
       "      <td>pubmed:22004763|imex:IM-16791|mint:MINT-820698...</td>\n",
       "      <td>taxid:11676(9hiv1)|taxid:11676(Human immunodef...</td>\n",
       "      <td>taxid:11676(9hiv1)|taxid:11676(Human immunodef...</td>\n",
       "      <td>psi-mi:\"MI:0915\"(physical association)</td>\n",
       "      <td>psi-mi:\"MI:0471\"(MINT)</td>\n",
       "      <td>intact:EBI-7484965|mint:MINT-8207016|imex:IM-1...</td>\n",
       "      <td>intact-miscore:0.40</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>uniprotkb:Q90VU7</td>\n",
       "      <td>uniprotkb:Q900A7</td>\n",
       "      <td>intact:EBI-7460704|intact:MINT-5281653</td>\n",
       "      <td>intact:EBI-7460739|intact:MINT-8208185</td>\n",
       "      <td>psi-mi:q90vu7_9hiv1(display_long)|uniprotkb:ne...</td>\n",
       "      <td>psi-mi:q900a7_9hiv1(display_long)|uniprotkb:ta...</td>\n",
       "      <td>psi-mi:\"MI:0416\"(fluorescence microscopy)</td>\n",
       "      <td>Sugiyama et al. (2011)</td>\n",
       "      <td>pubmed:21970979|imex:IM-16787|mint:MINT-820620...</td>\n",
       "      <td>taxid:11676(9hiv1)|taxid:11676(Human immunodef...</td>\n",
       "      <td>taxid:11676(9hiv1)|taxid:11676(Human immunodef...</td>\n",
       "      <td>psi-mi:\"MI:0403\"(colocalization)</td>\n",
       "      <td>psi-mi:\"MI:0471\"(MINT)</td>\n",
       "      <td>intact:EBI-7460732|mint:MINT-8206271|imex:IM-1...</td>\n",
       "      <td>intact-miscore:0.27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>uniprotkb:Q2Q067</td>\n",
       "      <td>uniprotkb:Q13015</td>\n",
       "      <td>intact:EBI-9675545</td>\n",
       "      <td>intact:EBI-6269719</td>\n",
       "      <td>psi-mi:q2q067_9dela(display_long)|uniprotkb:HB...</td>\n",
       "      <td>psi-mi:af1q_human(display_long)|uniprotkb:MLLT...</td>\n",
       "      <td>psi-mi:\"MI:0397\"(two hybrid array)</td>\n",
       "      <td>Simonis et al. (2012)</td>\n",
       "      <td>imex:IM-22977|pubmed:22458338</td>\n",
       "      <td>taxid:11908(humt-)|taxid:11908(Human T-lymphot...</td>\n",
       "      <td>taxid:9606(human)|taxid:9606(Homo sapiens)</td>\n",
       "      <td>psi-mi:\"MI:0915\"(physical association)</td>\n",
       "      <td>psi-mi:\"MI:0471\"(MINT)</td>\n",
       "      <td>intact:EBI-9675551|imex:IM-22977-4</td>\n",
       "      <td>intact-miscore:0.49</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                 0                 1                                       2   \\\n",
       "0  uniprotkb:Q72547  uniprotkb:Q72547  intact:EBI-7484755|intact:MINT-8208544   \n",
       "1  uniprotkb:Q72547  uniprotkb:Q72547  intact:EBI-7484755|intact:MINT-8208544   \n",
       "2  uniprotkb:Q72547  uniprotkb:Q72547  intact:EBI-7484755|intact:MINT-8208544   \n",
       "3  uniprotkb:Q90VU7  uniprotkb:Q900A7  intact:EBI-7460704|intact:MINT-5281653   \n",
       "4  uniprotkb:Q2Q067  uniprotkb:Q13015                      intact:EBI-9675545   \n",
       "\n",
       "                                       3   \\\n",
       "0  intact:EBI-7484755|intact:MINT-8208544   \n",
       "1  intact:EBI-7484755|intact:MINT-8208544   \n",
       "2  intact:EBI-7484755|intact:MINT-8208544   \n",
       "3  intact:EBI-7460739|intact:MINT-8208185   \n",
       "4                      intact:EBI-6269719   \n",
       "\n",
       "                                                  4   \\\n",
       "0  psi-mi:q72547_9hiv1(display_long)|uniprotkb:po...   \n",
       "1  psi-mi:q72547_9hiv1(display_long)|uniprotkb:po...   \n",
       "2  psi-mi:q72547_9hiv1(display_long)|uniprotkb:po...   \n",
       "3  psi-mi:q90vu7_9hiv1(display_long)|uniprotkb:ne...   \n",
       "4  psi-mi:q2q067_9dela(display_long)|uniprotkb:HB...   \n",
       "\n",
       "                                                  5   \\\n",
       "0  psi-mi:q72547_9hiv1(display_long)|uniprotkb:po...   \n",
       "1  psi-mi:q72547_9hiv1(display_long)|uniprotkb:po...   \n",
       "2  psi-mi:q72547_9hiv1(display_long)|uniprotkb:po...   \n",
       "3  psi-mi:q900a7_9hiv1(display_long)|uniprotkb:ta...   \n",
       "4  psi-mi:af1q_human(display_long)|uniprotkb:MLLT...   \n",
       "\n",
       "                                          6                         7   \\\n",
       "0      psi-mi:\"MI:0030\"(cross-linking study)  Nishitsuji et al. (2011)   \n",
       "1      psi-mi:\"MI:0030\"(cross-linking study)  Nishitsuji et al. (2011)   \n",
       "2      psi-mi:\"MI:0030\"(cross-linking study)  Nishitsuji et al. (2011)   \n",
       "3  psi-mi:\"MI:0416\"(fluorescence microscopy)    Sugiyama et al. (2011)   \n",
       "4         psi-mi:\"MI:0397\"(two hybrid array)     Simonis et al. (2012)   \n",
       "\n",
       "                                                  8   \\\n",
       "0  pubmed:22004763|imex:IM-16791|mint:MINT-820698...   \n",
       "1  pubmed:22004763|imex:IM-16791|mint:MINT-820698...   \n",
       "2  pubmed:22004763|imex:IM-16791|mint:MINT-820698...   \n",
       "3  pubmed:21970979|imex:IM-16787|mint:MINT-820620...   \n",
       "4                      imex:IM-22977|pubmed:22458338   \n",
       "\n",
       "                                                  9   \\\n",
       "0  taxid:11676(9hiv1)|taxid:11676(Human immunodef...   \n",
       "1  taxid:11676(9hiv1)|taxid:11676(Human immunodef...   \n",
       "2  taxid:11676(9hiv1)|taxid:11676(Human immunodef...   \n",
       "3  taxid:11676(9hiv1)|taxid:11676(Human immunodef...   \n",
       "4  taxid:11908(humt-)|taxid:11908(Human T-lymphot...   \n",
       "\n",
       "                                                  10  \\\n",
       "0  taxid:11676(9hiv1)|taxid:11676(Human immunodef...   \n",
       "1  taxid:11676(9hiv1)|taxid:11676(Human immunodef...   \n",
       "2  taxid:11676(9hiv1)|taxid:11676(Human immunodef...   \n",
       "3  taxid:11676(9hiv1)|taxid:11676(Human immunodef...   \n",
       "4         taxid:9606(human)|taxid:9606(Homo sapiens)   \n",
       "\n",
       "                                       11                      12  \\\n",
       "0  psi-mi:\"MI:0915\"(physical association)  psi-mi:\"MI:0471\"(MINT)   \n",
       "1  psi-mi:\"MI:0915\"(physical association)  psi-mi:\"MI:0471\"(MINT)   \n",
       "2  psi-mi:\"MI:0915\"(physical association)  psi-mi:\"MI:0471\"(MINT)   \n",
       "3        psi-mi:\"MI:0403\"(colocalization)  psi-mi:\"MI:0471\"(MINT)   \n",
       "4  psi-mi:\"MI:0915\"(physical association)  psi-mi:\"MI:0471\"(MINT)   \n",
       "\n",
       "                                                  13                   14  \n",
       "0  intact:EBI-7484777|mint:MINT-8206988|imex:IM-1...  intact-miscore:0.40  \n",
       "1  intact:EBI-7484916|mint:MINT-8207003|imex:IM-1...  intact-miscore:0.40  \n",
       "2  intact:EBI-7484965|mint:MINT-8207016|imex:IM-1...  intact-miscore:0.40  \n",
       "3  intact:EBI-7460732|mint:MINT-8206271|imex:IM-1...  intact-miscore:0.27  \n",
       "4                 intact:EBI-9675551|imex:IM-22977-4  intact-miscore:0.49  "
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mint = pd.read_csv('../data/networkInputs/species_human.txt', sep='\\t', header=None)\n",
    "\n",
    "# drop empty interaction columns\n",
    "mint = mint[mint[0].apply(lambda x: ':' in x)]\n",
    "mint = mint[mint[1].apply(lambda x: ':' in x)]\n",
    "\n",
    "data_sources['mint'] = mint\n",
    "data_preproc['mint'] = {\n",
    "    'columns': [0,1],\n",
    "    'maps':[\n",
    "        lambda x: id_dict(x.split(':')[1]),\n",
    "        lambda x: id_dict(x.split(':')[1])\n",
    "    ]\n",
    "}\n",
    "\n",
    "print(mint.columns)\n",
    "mint.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "#!wget https://stringdb-static.org/download/protein.links.full.v11.0/9606.protein.links.full.v11.0.txt.gz\n",
    "#!mv 9606.protein.links.full.v11.0.txt.gz ../data/networkInputs/\n",
    "#!gunzip ../data/networkInputs/9606.protein.links.full.v11.0.txt.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Index(['protein1', 'protein2', 'neighborhood', 'neighborhood_transferred',\n",
      "       'fusion', 'cooccurence', 'homology', 'coexpression',\n",
      "       'coexpression_transferred', 'experiments', 'experiments_transferred',\n",
      "       'database', 'database_transferred', 'textmining',\n",
      "       'textmining_transferred', 'combined_score'],\n",
      "      dtype='object')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>protein1</th>\n",
       "      <th>protein2</th>\n",
       "      <th>neighborhood</th>\n",
       "      <th>neighborhood_transferred</th>\n",
       "      <th>fusion</th>\n",
       "      <th>cooccurence</th>\n",
       "      <th>homology</th>\n",
       "      <th>coexpression</th>\n",
       "      <th>coexpression_transferred</th>\n",
       "      <th>experiments</th>\n",
       "      <th>experiments_transferred</th>\n",
       "      <th>database</th>\n",
       "      <th>database_transferred</th>\n",
       "      <th>textmining</th>\n",
       "      <th>textmining_transferred</th>\n",
       "      <th>combined_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>9606.ENSP00000000233</td>\n",
       "      <td>9606.ENSP00000432568</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>62</td>\n",
       "      <td>0</td>\n",
       "      <td>63</td>\n",
       "      <td>900</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>94</td>\n",
       "      <td>909</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>9606.ENSP00000000233</td>\n",
       "      <td>9606.ENSP00000427900</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>131</td>\n",
       "      <td>900</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>56</td>\n",
       "      <td>910</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>9606.ENSP00000000233</td>\n",
       "      <td>9606.ENSP00000354878</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>131</td>\n",
       "      <td>900</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>56</td>\n",
       "      <td>910</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52</th>\n",
       "      <td>9606.ENSP00000000233</td>\n",
       "      <td>9606.ENSP00000405926</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>63</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>900</td>\n",
       "      <td>0</td>\n",
       "      <td>64</td>\n",
       "      <td>58</td>\n",
       "      <td>906</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78</th>\n",
       "      <td>9606.ENSP00000000233</td>\n",
       "      <td>9606.ENSP00000314615</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>61</td>\n",
       "      <td>63</td>\n",
       "      <td>0</td>\n",
       "      <td>262</td>\n",
       "      <td>900</td>\n",
       "      <td>0</td>\n",
       "      <td>473</td>\n",
       "      <td>317</td>\n",
       "      <td>971</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                protein1              protein2  neighborhood  \\\n",
       "15  9606.ENSP00000000233  9606.ENSP00000432568             0   \n",
       "16  9606.ENSP00000000233  9606.ENSP00000427900             0   \n",
       "40  9606.ENSP00000000233  9606.ENSP00000354878             0   \n",
       "52  9606.ENSP00000000233  9606.ENSP00000405926             0   \n",
       "78  9606.ENSP00000000233  9606.ENSP00000314615             0   \n",
       "\n",
       "    neighborhood_transferred  fusion  cooccurence  homology  coexpression  \\\n",
       "15                         0       0            0         0             0   \n",
       "16                         0       0            0         0             0   \n",
       "40                         0       0            0         0             0   \n",
       "52                         0       0            0         0             0   \n",
       "78                         0       0            0         0            61   \n",
       "\n",
       "    coexpression_transferred  experiments  experiments_transferred  database  \\\n",
       "15                        62            0                       63       900   \n",
       "16                         0            0                      131       900   \n",
       "40                         0            0                      131       900   \n",
       "52                        63            0                        0       900   \n",
       "78                        63            0                      262       900   \n",
       "\n",
       "    database_transferred  textmining  textmining_transferred  combined_score  \n",
       "15                     0           0                      94             909  \n",
       "16                     0           0                      56             910  \n",
       "40                     0           0                      56             910  \n",
       "52                     0          64                      58             906  \n",
       "78                     0         473                     317             971  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Int64Index([0, 1, 2, 3, 4], dtype='int64')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>9606</td>\n",
       "      <td>Q03135|CAV1_HUMAN</td>\n",
       "      <td>9606.ENSP00000339191</td>\n",
       "      <td>100.0</td>\n",
       "      <td>370.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>9606</td>\n",
       "      <td>P35125|UBP6_HUMAN</td>\n",
       "      <td>9606.ENSP00000460380</td>\n",
       "      <td>100.0</td>\n",
       "      <td>2941.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>9606</td>\n",
       "      <td>P41161|ETV5_HUMAN</td>\n",
       "      <td>9606.ENSP00000306894</td>\n",
       "      <td>100.0</td>\n",
       "      <td>1045.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>9606</td>\n",
       "      <td>Q6KF10|GDF6_HUMAN</td>\n",
       "      <td>9606.ENSP00000287020</td>\n",
       "      <td>100.0</td>\n",
       "      <td>928.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>9606</td>\n",
       "      <td>Q495C1|RN212_HUMAN</td>\n",
       "      <td>9606.ENSP00000389709</td>\n",
       "      <td>100.0</td>\n",
       "      <td>610.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      0                   1                     2      3       4\n",
       "0  9606   Q03135|CAV1_HUMAN  9606.ENSP00000339191  100.0   370.0\n",
       "1  9606   P35125|UBP6_HUMAN  9606.ENSP00000460380  100.0  2941.0\n",
       "2  9606   P41161|ETV5_HUMAN  9606.ENSP00000306894  100.0  1045.0\n",
       "3  9606   Q6KF10|GDF6_HUMAN  9606.ENSP00000287020  100.0   928.0\n",
       "4  9606  Q495C1|RN212_HUMAN  9606.ENSP00000389709  100.0   610.0"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "string_database_score_filter = 900\n",
    "\n",
    "string = pd.read_csv('../data/networkInputs/9606.protein.links.full.v11.0.txt', sep=' ')\n",
    "string = string[string['database']>=string_database_score_filter]\n",
    "string_alias = pd.read_csv('../data/networkInputs/human.uniprot_2_string.2018.tsv', sep='\\t', header=None)\n",
    "\n",
    "string_to_uniprot = dict(zip(string_alias[2], string_alias[1].apply(lambda x: x.split('|')[1].split('_')[0])))\n",
    "\n",
    "data_sources['string'] = string\n",
    "data_preproc['string'] = {\n",
    "    'columns': ['protein1','protein2'],\n",
    "    'maps': [\n",
    "        lambda x: id_dict(string_to_uniprot[x]) if x in string_to_uniprot.keys() else None,\n",
    "        lambda x: id_dict(string_to_uniprot[x]) if x in string_to_uniprot.keys() else None\n",
    "    ]\n",
    "}\n",
    "\n",
    "print(string.columns)\n",
    "display(string.head())\n",
    "print(string_alias.columns)\n",
    "display(string_alias.head())"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Network Assembly"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Node1</th>\n",
       "      <th>Node2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [Node1, Node2]\n",
       "Index: []"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "network = pd.DataFrame(columns=['Node1', 'Node2'])\n",
    "network.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Adding hippie to network\n",
      "Adding phosphosite_substrate to network\n",
      "Adding hprd to network\n",
      "Adding reactome to network\n",
      "Adding biogrid to network\n",
      "Adding i2d to network\n",
      "Adding mint to network\n",
      "Adding string to network\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "(5066, 2)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Node1</th>\n",
       "      <th>Node2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MST1R</td>\n",
       "      <td>YES1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TYRO3</td>\n",
       "      <td>YES1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>FGR</td>\n",
       "      <td>SRC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>ATR</td>\n",
       "      <td>FLT1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>LYN</td>\n",
       "      <td>PRKCD</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    Node1  Node2\n",
       "2   MST1R   YES1\n",
       "3   TYRO3   YES1\n",
       "11    FGR    SRC\n",
       "16    ATR   FLT1\n",
       "19    LYN  PRKCD"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "network = pd.DataFrame(columns=['Node1', 'Node2'])\n",
    "\n",
    "for source_name, data in data_sources.items():\n",
    "    \n",
    "    #if source_name in ['hippie', 'phosphosite_substrate', 'hprd', 'biogrid']:\n",
    "    #    continue\n",
    "    \n",
    "    print('Adding', source_name, 'to network')\n",
    "    \n",
    "    left_col, right_col = data_preproc[source_name]['columns']\n",
    "    left_map, right_map = data_preproc[source_name]['maps']\n",
    "    \n",
    "    data = data[[left_col, right_col]].dropna(axis=0)\n",
    "    \n",
    "    #print(data.dtypes)#[left_col])\n",
    "    \n",
    "    to_add = pd.DataFrame(columns=network.columns)\n",
    "    to_add['Node1'] = data[left_col].apply(left_map)\n",
    "    to_add['Node2'] = data[right_col].apply(right_map)\n",
    "    to_add.dropna(axis=0, inplace = True)\n",
    "   \n",
    "    network = pd.concat([network, to_add], ignore_index=True)\n",
    "    \n",
    "    #break\n",
    "    \n",
    "# drop obvious duplicates\n",
    "network.drop_duplicates(inplace = True)\n",
    "\n",
    "# drop items not in the target kinome set\n",
    "network = network[(network['Node1'].isin(kinome_set)) & (network['Node2'].isin(kinome_set))]\n",
    "\n",
    "network_cop = copy(network[['Node2','Node1']])\n",
    "network_cop.columns = ['Node1', 'Node2']\n",
    "\n",
    "# drop a -> b b -> a redundancies \n",
    "network = network.append(network_cop, ignore_index=True).drop_duplicates()\n",
    "network = network[network['Node1']<network['Node2']]\n",
    "\n",
    "#for i in network.index:\n",
    "#    network.loc[i] = sorted(network.loc[i])\n",
    "    \n",
    "#network.drop_duplicates(inplace=True)\n",
    "\n",
    "display(network.shape)\n",
    "network.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "network.to_csv('../data/kin/kin_unweighted.csv', sep='\\t', header=None, index=False)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python [conda env:anaconda3]",
   "language": "python",
   "name": "conda-env-anaconda3-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
